xwayland/xwm.c

```diff
@@ -1604,10 +1639,10 @@ void xwm_destroy(struct wlr_xwm *xwm) {
 	}
 #endif
 	struct wlr_xwayland_surface *xsurface, *tmp;
-	wl_list_for_each_safe(xsurface, tmp, &xwm->surfaces, link) {
+	wl_list_for_each_safe(xsurface, tmp, &xwm->surfaces_in_map_order, map_link) {
 		xwayland_surface_destroy(xsurface);
 	}
-	wl_list_for_each_safe(xsurface, tmp, &xwm->unpaired_surfaces, link) {
+	wl_list_for_each_safe(xsurface, tmp, &xwm->unpaired_surfaces, unpaired_link) {
```

Comment:
Note for reviewer: this change isn't related to this patch, but it seemed broken so I changed it anyway.

Reply:
Can we isolate this change into a separate commit? That way we can safely revert other commits if necessary.

Reply:
Done.